From 125e65650fbed0a5788e75ea2a6bb24dd2eec11d Mon Sep 17 00:00:00 2001
From: Jing Ji <jji@google.com>
Date: Mon, 4 Nov 2019 14:22:27 -0800
Subject: [PATCH 3/3] Prevent system uid component from running in an isolated
 app process

Bug: 140055304
Test: Manua
Change-Id: Ie7f6ed23f0c6009aad0f67a00af119b02cdceac3
Merged-In: I5a1618fab529cb0300d4a8e9c7762ee218ca09eb
(cherry picked from commit 0bfebadf304bdd5f921e80f93de3e0d13b88b79c)
---
 .../java/com/android/server/am/ActivityManagerService.java     | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 451c2ffb6c2..aa6f124d63c 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -3467,7 +3467,8 @@ public final class ActivityManagerService extends ActivityManagerNative
             final int procCount = procs.size();
             for (int i = 0; i < procCount; i++) {
                 final int procUid = procs.keyAt(i);
-                if (UserHandle.isApp(procUid) || !UserHandle.isSameUser(procUid, uid)) {
+                if (UserHandle.isApp(procUid) || !UserHandle.isSameUser(procUid, uid)
+                        || UserHandle.isIsolated(procUid)) {
                     // Don't use an app process or different user process for system component.
                     continue;
                 }
-- 
2.17.1

